lib/event_tracer/prometheus_logger.rb

```diff
@@ -0,0 +1,97 @@
+module EventTracer
+  class PrometheusLogger < BasicDecorator
+
+    SUPPORTED_METRIC_TYPES = {
+      counter: :increment_count,
+      gauge: :set_gauge
+    }.freeze
+    DEFAULT_INCREMENT = 1
+
+    attr_reader :allowed_tags
+
+    def initialize(prometheus, allowed_tags: [], default_tags: {}, raise_if_missing: true)
+      super(prometheus)
+      @allowed_tags = allowed_tags.freeze
+      @default_tags = default_tags.freeze
+      @raise_if_missing = raise_if_missing
+    end
+
+    LOG_TYPES.each do |log_type|
+      define_method log_type do |**args|
```

Comment:
Appsignal, Datadog and this logger seems to have very similar process here. Could we add a task to refactor them?

Reply:
Yes sir I'll add a task for it